hoomd/md/pytest/test_rigid.py

```diff
@@ -453,3 +453,152 @@ def test_rigid_dof(
+    integrator = md.Integrator(
+        dt=0.005,
+        methods=[constvol],
+        integrate_rotational_dof=True,
+    )
+    integrator.rigid = rigid
+    sim.operations += integrator
+    sim.run(1)
+
+    new_snapshot = sim.state.get_snapshot()
+    expected_velocity = [com_velocity] * 5
+    assert np.allclose(expected_velocity, new_snapshot.particles.velocity)
+
+    constant_torque = hoomd.md.force.Constant(filter=hoomd.filter.All())
+    constant_torque.constant_torque["A"] = (0, 0, 1)
+    integrator.forces.append(constant_torque)
+
+    sim.run(1)
+    # compare the resulting tangential velocities with expected values
+    expected_tangential_velocity = [
+        [0, 0, 0],
+        [-6.25e-08, 5.00e-03, 0.00e00],
+        [6.25e-08, -5.00e-03, 0.00e00],
+        [5.00e-03, 6.25e-08, 0.00e00],
+        [-5.00e-03, -6.25e-08, 0.00e00],
+    ]
+    expected_velocity = np.add(expected_tangential_velocity, com_velocity)
+    new_snapshot = sim.state.get_snapshot()
+    assert np.allclose(expected_velocity, new_snapshot.particles.velocity)
+
+
+@pytest.mark.parametrize(
+    "omega",
+    [
+        [10, 0, 0],
+        [0, 10, 0],
+        [0, 0, 10],
+        [3, 4, 10],
+    ],
+)
+@pytest.mark.parametrize(
+    "body_definition,moment_of_inertia,mass",
+    [
+        (
+            {
+                "constituent_types": ["B", "B", "B", "B"],
+                "positions": [
+                    [1, 0, -1 / (2 ** (1.0 / 2.0))],
+                    [-1, 0, -1 / (2 ** (1.0 / 2.0))],
+                    [0, -1, 1 / (2 ** (1.0 / 2.0))],
+                    [0, 1, 1 / (2 ** (1.0 / 2.0))],
+                ],
+                "orientations": [(1.0, 0.0, 0.0, 0.0)] * 4,
+            },
+            [1, 1, 1],
+            [1],
+        ),
+        (
+            {
+                "constituent_types": ["B", "B"],
+                "positions": [[-1.2, 0, 0], [1.2, 0, 0]],
+                "orientations": [[1, 0, 0, 0], [1, 0, 0, 0]],
+            },
+            [0.0, 2.88, 2.88],
+            [2],
+        ),
+    ],
+    ids=["simple-inertia-body", "dimer-body"],
+)
+@pytest.mark.parametrize(
+    "com_velocity",
+    [(0, 0, 0), (1, 2, 3)],
+)
+def test_velocity_constituents_constant_angmom(
+    omega,
+    mass,
+    com_velocity,
+    body_definition,
+    moment_of_inertia,
+    simulation_factory,
+    one_particle_snapshot_factory,
+):
+    """Test velocity of constituents updates correctly with 0 timestep."""
+    initial_snapshot = one_particle_snapshot_factory(particle_types=["A", "B"])
+    # calculate angular momentum quaternion from desired angular velocity
+    # steps are typically the following:
+    # 1. Rotate space-frame angular velocity to body frame
+    # 2. Multiply by moment of inertia tensor in body frame
+    # 3. Promote to a quaternion with 0 real part and multiply by orientation
+    # 4. Multiply by 2
+    angmom = np.zeros(4)
+    angmom[1:] = np.multiply(omega, moment_of_inertia) * 2
```

Comment:
I misread this as a square. It's conventional to put constants first.
```suggestion
    angmom[1:] = 2 * np.multiply(omega, moment_of_inertia)
```